pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -989,17 +1000,33 @@ func (a *ApplicationBackupController) backupResources(
 	allObjects := make([]runtime.Unstructured, 0)
 	for i := 0; i < len(namespacelist); i += backupResourcesBatchCount {
 		batch := namespacelist[i:min(i+backupResourcesBatchCount, len(namespacelist))]
-		objects, err := a.resourceCollector.GetResources(
-			batch,
-			backup.Spec.Selectors,
-			objectMap,
-			optionalBackupResources,
-			true)
-		if err != nil {
-			log.ApplicationBackupLog(backup).Errorf("Error getting resources: %v", err)
-			return err
+		if len(backup.Spec.ResourceTypes) == 0 {
+			batch := namespacelist[i:min(i+backupResourcesBatchCount, len(namespacelist))]
```

Comment:
This batch can be removed as we have one outside the if condition

Reply:
Yeah, overlooked it. Fixed it.